vagrant/base.sh

```diff
@@ -20,38 +20,43 @@ set -ex
 # If you update this, also update tests/docker/Dockerfile
 export KIBOSH_VERSION=d85ac3ec44be0700efe605c16289fd901cfdaa13
 
-if [ -z `which javac` ]; then
-    apt-get -y update
-    apt-get install -y software-properties-common python-software-properties
-    add-apt-repository -y ppa:webupd8team/java
-    apt-get -y update
+path_to_jdk_cache() {
+  jdk_version=$1
+  echo "/tmp/jdk-${jdk_version}.tar.gz"
+}
 
-    # Try to share cache. See Vagrantfile for details
-    mkdir -p /var/cache/oracle-jdk8-installer
-    if [ -e "/tmp/oracle-jdk8-installer-cache/" ]; then
-        find /tmp/oracle-jdk8-installer-cache/ -not -empty -exec cp '{}' /var/cache/oracle-jdk8-installer/ \;
-    fi
-    if [ ! -e "/var/cache/oracle-jdk8-installer/jdk-8u171-linux-x64.tar.gz" ]; then
-      # Grab a copy of the JDK since it has moved and original downloader won't work
-      curl -s -L "https://s3-us-west-2.amazonaws.com/kafka-packages/jdk-8u171-linux-x64.tar.gz" -o /var/cache/oracle-jdk8-installer/jdk-8u171-linux-x64.tar.gz
-    fi
+fetch_jdk_tgz() {
+  jdk_version=$1
 
-    /bin/echo debconf shared/accepted-oracle-license-v1-1 select true | /usr/bin/debconf-set-selections
+  path=$(path_to_jdk_cache $jdk_version)
 
-    # oracle-javaX-installer runs wget with a dot progress indicator which ends up
-    # as one line per dot in the build logs.
-    # To avoid this noise we redirect all output to a file that we only show if apt-get fails.
-    echo "Installing JDK..."
-    if ! apt-get -y install oracle-java8-installer oracle-java8-set-default >/tmp/jdk_install.log 2>&1 ; then
-        cat /tmp/jdk_install.log
-        echo "ERROR: JDK install failed"
-        exit 1
-    fi
+  if [ ! -e $path ]; then
+    mkdir -p $(dirname $path)
+    curl -s -L "https://s3-us-west-2.amazonaws.com/kafka-packages/jdk-${jdk_version}.tar.gz" -o $path
+  fi
+}
+
+JDK_MAJOR="${JDK_MAJOR:-8}"
+JDK_FULL="${JDK_FULL:-8u202-linux-x64}"
+
+if [ -z `which javac` ]; then
+    apt-get -y update
+    apt-get install -y software-properties-common python-software-properties binutils java-common
+
+    echo "===> Installing JDK..." 
+
+    mkdir -p /opt/jdk
+    cd /opt/jdk
```

Comment:
i think subsequent `tar` command is leaving the entire jdk in `/opt/jdk` without a jdk version directory. this line and the line above should include `$JDK_MAJOR` as a directory.